pyrobosim/pyrobosim/navigation/a_star.py

```diff
@@ -158,13 +165,18 @@ def __init__(self, **planner_config):
         super().__init__()
         self.impl = None  # Holds the implementation.
 
-        # Depending on if grid is provided, select the implementation.
-        if planner_config.get("grid", None):
-            self.impl = AStarGrid(**planner_config)
-        else:
-            raise NotImplementedError(
-                "A-star does not have a standalone graph based implementation."
-            )
+        if not hasattr(planner_config["robot"], "world") or planner_config["robot"].world is None:
+            raise AttributeError("Planner expects the robot to have a `world` attribute")
+            
+        self.impl = AStarGrid(**planner_config)
```

Comment:
Are we able to / do you plan to get rid of this `impl` pattern now? Seems like just having a flat list of planners could work.